jgiven-html-app/src/js/util.js

```diff
@@ -144,3 +144,65 @@ export function deselectAll (options) {
     option.selected = false;
   });
 }
+
+function getArgumentList(wordList) {
+  function isArgument(word) { return !!word.argumentInfo; }
+  return _.chain(wordList)
+          .filter(isArgument)
+          .map(function (word) { return word.value; })
+          .value();
+}
+
+// can be removed if lodash-version is >= 3.10.1
+_.takeWhile = function (arr, predicate) {
+    var predicateHoldsForAll = true;
+    return _.filter(arr, function (e) {
+        var predicateResult = predicate(e);
+        predicateHoldsForAll = !predicateResult ? false : predicateHoldsForAll;
+        return predicateResult && predicateHoldsForAll;
+    });
+};
+
+function parseNextInt(arr) {
+    var numbers = _.takeWhile(arr, function (c) { return !isNaN(c) && c !== " "; }).join("");
+    var parsedInt = parseInt(numbers);
+    var result  = { integer : isNaN(parsedInt) ? undefined : parsedInt
+                  , length  : numbers.length } ;
+    return result;
+}
+
+function replaceArguments(string, argumentList) {
+    var separator = "$";
+    var result    = [];
+    var placeHolderCount = 0;
+
+    for(var i = 0; i < string.length; ++i) {
+        var c           = string.charAt(i);
```

Comment:
I am not a big fan of aligning code vertically. I typically automatically format my code, so this hand-crafted formatting will be destroyed the next time I touch the file. I guess we need a coding style guide for the JavaScript code :-)